lint/base_linter/python_linter.py

```diff
@@ -1,314 +1,330 @@
 """This module exports the PythonLinter subclass of Linter."""
 
-import importlib
+from functools import lru_cache
 import os
 import re
+import subprocess
 
+import sublime
 from .. import linter, persist, util
 
 
 class PythonLinter(linter.Linter):
-    """
-    This Linter subclass provides python-specific functionality.
-
-    Linters that check python should inherit from this class.
-    By doing so, they automatically get the following features:
-
-    - comment_re is defined correctly for python.
-
-    - A python shebang is returned as the @python:<version> meta setting.
-
-    - Execution directly via a module method or via an executable.
-
-    If the module attribute is defined and is successfully imported,
-    whether it is used depends on the following algorithm:
-
-      - If the cmd attribute specifies @python and ST's python
-        satisfies that version, the module will be used. Note that this
-        check is done during class construction.
-
-      - If the check_version attribute is False, the module will be used
-        because the module is not version-sensitive.
-
-      - If the "@python" setting is set and ST's python satisfies
-        that version, the module will be used.
-
-      - Otherwise the executable will be used with the python specified
-        in the "@python" setting, the cmd attribute, or the default system
-        python.
-
-    """
-
-    SHEBANG_RE = re.compile(r'\s*#!(?:(?:/[^/]+)*[/ ])?python(?P<version>\d(?:\.\d)?)')
+    """(New) Python Linter [WIP]."""
```

Comment:
Is it still WIP or are you ready for this to get merged? It seems to work fine on my machine.... We can always make fixes and adjustments later, but I'd like to expose it to a slightly wider audience.

Reply:
Done